device/mockup/tt_mockup_device.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+#include "device/tt_cluster_descriptor.h"
+#include "device/tt_device.h"
+
+class tt_MockupDevice : public tt_device {
+   public:
+    tt_MockupDevice(const std::string& sdesc_path) : tt_device(sdesc_path) {
+        soc_descriptor_per_chip.emplace(0, tt_SocDescriptor(sdesc_path));
+        std::set<chip_id_t> target_devices = {0};
+    }
+    virtual ~tt_MockupDevice() {}
+
+    // Setup/Teardown Functions
+    virtual std::unordered_map<chip_id_t, tt_SocDescriptor>& get_virtual_soc_descriptors() override {
+        return soc_descriptor_per_chip;
+    }
+    virtual void set_device_l1_address_params(const tt_device_l1_address_params& l1_address_params_) override{};
```

Comment:
nit: virtual is not needed when your implementation is final. So no "virtual" needed in this whole file.